src/include/crt_utils.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * (C) Copyright 2019-2021 Intel Corporation.
```

Comment:
This is very weird.   We are installing a library but no header file to actually use that library.  In fact, all headers are internal.  The library is not internal because it's used by self_test and crt_ctl

This should probably go in cart/util.h and should remove any references to "crt_internal.h"

Reply:
We have installed an include file to use the library, crt_utils.h  
However, we cannot build an executable on an rpm because the crt_internal.h is not included in the release.  If an executable is pre-built for the rpm, then the library is required.
The big issue seems to be utils/ctl and utils/self_test.  These seems to have started as cart testing programs that were moved into the utils directory.  However, they included multiple internal cart testing include files and these dependencies were never removed (crt_internal.h just includes all of the other include files).  To remove crt_internal.h from crt_utils,h would require a  rework of  ctl and self_test

Reply:
why would it require any rework?  Worst case scenario is you need to include crt_internal.h in the .c files rather than in the header.

Regardless, I think either an internal static library (with -fPIC) or the trick I mentioned in chat where we do something like

crt_utils = env.SharedObject([files])
Export('crt_utils')

And then adding crt_utils where we need it is better than shipping a library.

Reply:
It would require re-work because ctl.c and self_test.c did NOT just follow the API interface.  Rather, they used internal cart tests enum and data structures to build their data structures to send data between nodes.
This is what I meant when I said it look as if they were originally internal cart tests that were moved to the utility directory.